pandas/core/dtypes/common.py

```diff
@@ -737,5 +737,7 @@ def pandas_dtype(dtype):
             pass
     elif isinstance(dtype, ExtensionDtype):
         return dtype
+    elif np.dtype(dtype).kind == 'O':
```

Comment:
`np.dtype` doesn't differentiate between [1] and [2], any thoughts on how to raise errors for [1] without breaking [2]? Since the change I made would raise an error in both cases.

[1]
```
>>> np.dtype(pd.Timestamp)
dtype('O')
```

[2]
```
>>> np.dtype('object')
dtype('O')
```